### PR TITLE
Update RVO2 to git 2022.09 (reverted)

### PR DESCRIPTION
**Pull request by DeeJayLSP**
I don't know if the "heavy modification for Godot" was done correctly, but at least it seems to build and work as it should.

I also touched the module.

There's a small probability that I missed some comments.

### CODE REVIEW

**Comment by smix8** [comment]
The modifications that @DeeJayLSP did seem fine. The RVO test projects so far run as expected with this newer, modified version.

That said I made no further tests and I do not spot anything substantial in the changelog that would make the update meaningful for Godot. It is all noise from code-style and "Google style better" changes.

**Review by smix8 — APPROVED** [review]

File: modules/navigation/SCsub [hunk 1]
```diff
@@ -36,12 +36,7 @@ if env["builtin_recastnavigation"]:
 # RVO 2D Thirdparty source files
 if env["builtin_rvo2_2d"]:
     thirdparty_dir = "#thirdparty/rvo2/rvo2_2d/"
-    thirdparty_sources = [
-        "Agent2d.cpp",
-        "Obstacle2d.cpp",
-        "KdTree2d.cpp",
-        "RVOSimulator2d.cpp",
-    ]
+    thirdparty_sources = ["Agent2d.cc", "Obstacle2d.cc", "KdTree2d.cc", "Line.cc", "RVOSimulator2d.cc", "Vector2.cc"]
```
Comment (akien-mga) [inline modules/navigation/SCsub:39]: Please keep those with one per line. You need to use a trailing comma for the last item.

**Comment by akien-mga** [comment]
> I don't know if the "heavy modification for Godot" was done correctly, but at least it seems to build and work as it should.

What are those heavy modifications for Godot exactly? Can we track them as a patch file so that they can be applied on update easily?

File: thirdparty/README.md [hunk 2]
```diff
@@ -630,13 +630,13 @@ Files extracted from upstream source:
 For 2D in `rvo2_2d` folder
 
 - Upstream: https://github.com/snape/RVO2
-- Version: git (f7c5380235f6c9ac8d19cbf71fc94e2d4758b0a3, 2021)
+- Version: git (a7a9575b24f6972f5edd0790784860406663c08f, 2023)
```
Comment (akien-mga) [inline thirdparty/README.md:633]: That commit is actually from 2022, not 2023.

There's a few commits in 2023 though, last one yesterday, might be worth syncing with those.

**Comment by DeeJayLSP** [comment]
Seems like snape is squashing commits, that's why hashes are different.

Also I'm pretty sure I did update those on readme.

Anyway, the last commit where the files we use were updated is from 2022.09 (for both). Is it OK to use that instead of the latest considering to avoid the squashes?

File: thirdparty/README.md (hunk 2 above)
Comment (DeeJayLSP) [reply thirdparty/README.md:633]: Probably because the latest release in the repo is from 2016

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Reverted for now to fix a regression (#78826). Let's work on doing this update again for 4.2.

**Pull request merged**